builtin/providers/azurerm/resource_arm_container_service.go

```diff
@@ -0,0 +1,669 @@
+	containerServiceClient := meta.(*ArmClient).containerServicesClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["containerServices"]
+
+	resp, err := containerServiceClient.Get(resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error making Read request on Azure Container Service %s: %s", name, err)
+	}
+	if resp.StatusCode == http.StatusNotFound {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+	d.Set("resource_group_name", resGroup)
+
+	d.Set("orchestration_platform", string(resp.Properties.OrchestratorProfile.OrchestratorType))
+
+	masterProfiles := flattenAzureRmContainerServiceMasterProfile(*resp.Properties.MasterProfile)
+	if masterProfiles != nil {
+		d.Set("master_profile", masterProfiles)
+	}
+
+	linuxProfile := flattenAzureRmContainerServiceLinuxProfile(*resp.Properties.LinuxProfile)
+	if linuxProfile != nil {
+		d.Set("linux_profile", linuxProfile)
+	}
+
+	agentPoolProfiles := flattenAzureRmContainerServiceAgentPoolProfiles(resp.Properties.AgentPoolProfiles)
+	if agentPoolProfiles != nil {
+		d.Set("agent_pool_profile", agentPoolProfiles)
+	}
+
+	servicePrincipal := flattenAzureRmContainerServiceServicePrincipalProfile(resp.Properties.ServicePrincipalProfile)
+	if servicePrincipal != nil {
+		d.Set("service_principal", servicePrincipal)
+	}
+
+	diagnosticProfile := flattenAzureRmContainerServiceDiagnosticsProfile(resp.Properties.DiagnosticsProfile)
+	if diagnosticProfile != nil {
+		d.Set("diagnostics_profile", diagnosticProfile)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmContainerServiceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	containerServiceClient := client.containerServicesClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["containerServices"]
+
+	resp, err := containerServiceClient.Delete(resGroup, name, make(chan struct{}))
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("Error issuing Azure ARM delete request of Container Service '%s': %s", name, err)
+	}
+
+	return nil
+
+}
+
+func flattenAzureRmContainerServiceMasterProfile(profile containerservice.MasterProfile) *schema.Set {
+	masterProfiles := &schema.Set{
+		F: resourceAzureRMContainerServiceMasterProfileHash,
+	}
+
+	masterProfile := make(map[string]interface{}, 2)
+
+	masterProfile["count"] = int(*profile.Count)
+	masterProfile["dns_prefix"] = *profile.DNSPrefix
+
+	masterProfiles.Add(masterProfile)
+
+	return masterProfiles
+}
+
+func flattenAzureRmContainerServiceLinuxProfile(profile containerservice.LinuxProfile) *schema.Set {
+	profiles := schema.Set{
+		F: resourceAzureRMContainerServiceLinuxProfilesHash,
+	}
+
+	values := map[string]interface{}{}
+
+	sshKeys := schema.Set{
```

Comment:
`ssh_key` is a TypeList, I think it needs an `[]interface{}` rather than a `schema.Set`.